server/src/main/java/com/paypal/selion/pojos/BrowserStatisticsCollection.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    /**
+     * Sets the maximum instances for a particular browser. This call creates a unique statistics for the provided
+     * browser name it does not exists.
+     * 
+     * @param browserName
+     *            Name of the browser.
+     * @param maxBrowserInstances
+     *            Maximum instances of the browser.
+     */
+    public void setMaxBrowserInstances(String browserName, int maxBrowserInstances) {
+        logger.entering(new Object[] { browserName, maxBrowserInstances });
+        validateBrowserName(browserName);
+        BrowserStatistics lStatistics = createStatisticsIfNotPresent(browserName);
+        lStatistics.setMaxBrowserInstances(maxBrowserInstances);
+        logger.exiting();
+    }
+
+    /**
+     * Increments the waiting request for the provided browser name. This call creates a unique statistics for the
+     * provided browser name it does not exists.
+     * 
+     * @param browserName
+     *            Name of the browser.
+     */
+    public void incrementWaitingRequests(String browserName) {
+        logger.entering(browserName);
+        validateBrowserName(browserName);
+        BrowserStatistics lStatistics = createStatisticsIfNotPresent(browserName);
+        lStatistics.incrementWaitingRequests();
+        logger.exiting();
+    }
+
+    private void validateBrowserName(String browserName) {
+        if (StringUtils.isBlank(browserName)) {
+            throw new IllegalArgumentException("Browser name cannot be null");
+        }
+    }
+
+    private synchronized BrowserStatistics createStatisticsIfNotPresent(String browserName) {
+        for (BrowserStatistics loadStatistics : this.browserStatisticsList) {
+            if (loadStatistics.browserName.equals(browserName)) {
+                return loadStatistics;
+            }
+        }
+        BrowserStatistics loadStatistics = new BrowserStatistics(browserName);
+        this.browserStatisticsList.add(loadStatistics);
+        return loadStatistics;
+    }
+
+    @Override
+    public String toString() {
+        Gson gson = new Gson();
+        return gson.toJson(browserStatisticsList);
+    }
+
+    /**
+     * <code>BrowserStatistics</code> holds the statistics for an individual browser. This class acts as a mere Value
+     * Object.
+     * 
+     */
+    private class BrowserStatistics {
+
+        private String browserName;
+
+        private Statistics statistics;
+
+        public BrowserStatistics(String browserName) {
+            this.browserName = browserName;
+            this.statistics = new Statistics();
+        }
+
+        public void setMaxBrowserInstances(int maxBrowserInstances) {
+            this.statistics.maxBrowserInstances = maxBrowserInstances;
+        }
+
+        public void incrementWaitingRequests() {
+            ++this.statistics.waitingRequests;
+        }
+
+        private class Statistics {
+
+            private int waitingRequests = 0;
+
+            private int maxBrowserInstances = 0;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid using redundant field initializer for 'maxBrowserInstances'](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103178712/issues/source?bid=2928425&fileBranchId=3143459#l126)